src/disk/header.rs

```diff
@@ -167,3 +167,411 @@ impl fmt::Debug for Header {
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00,
+    ];
+
+    /// Header block from MEGA65 disk "Mega Pixel Bomber Blitz" -- pixelblitz_vUSa4k.d81
+    /// https://files.mega65.org/html/main.php?id=9a6c653c-93de-41c0-b890-0ce56c39fde2
+    static HEADER_MEGA65_PIXELBLITZ_D81: [u8; 256] = [
+        0x28, 0x03, 0x44, 0x00, 0x50, 0x49, 0x58, 0x45, 0x4c, 0x20, 0x42, 0x4c, 0x49, 0x54, 0x5a,
+        0xa0, 0xa0, 0xa0, 0xa0, 0xa0, 0xa0, 0xa0, 0x32, 0x30, 0x32, 0x32, 0xa0, 0xa0, 0xa0, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00,
+    ];
+
+    /// Parse the provided header block according to the header format used by the
+    /// provided disk type. Panic on error.
+    fn parse_header(disk_type: DiskType, header_block: &[u8]) -> Header {
```

Comment:
Perhaps overkill for a test, but panicking could be avoided by using `-> Result<Header>` so that `.unwrap()` can be replaced with `?`.

Reply:
It is overkill for a test. :) Three possibilities come to mind for handling the unexpected case of `Header::read()` returning `Err(_)`:

1. **Use the existing code that `unwrap()`'s in `parse_header()`.** When run with `RUST_BACKTRACE=1` and examining the resulting stack trace after `Header::read()` returns `Err(_)`, it is very clear which line of `test_headers()` resulted in the issue, and it is also clear that the problem occurred within `parse_header()`, so any such issues should be quick to diagnose.

2. **Have `parse_header()` return `io::Result<Header>`, and `unwrap()` in the caller.** This doesn't eliminate the `unwrap()`, since it just moves it from one place to another. Stack traces would still be pretty good.

3. **Have `test_headers()` return `io::Result<()>` and propagate the error to the test framework via `?`.** This eliminates all `unwrap()`'s, but the resulting stack trace is very opaque about where the problem occurred. The only clue in the stack trace is that the problem occurred somewhere in the `test_headers()` function. It also doesn't really eliminate a panic, since the test framework will panic when we return the error.

Given the above, I'm leaning towards eagerly `unwrap()`'ing in tests when we truly expect the `Result` to be `Ok(_)`, so that if that expectation is not met, the maintainer is given high-resolution information about the issue.